primitives/arithmetic/src/fixed_point.rs

```diff
@@ -536,10 +536,10 @@ macro_rules! implement_fixed {
 			}
 		}
 
-		impl<P: PerThing> From<P> for $name {
+		impl<P: PerThing> From<P> for $name where P::Inner: FixedPointOperand {
 			fn from(p: P) -> Self {
-				let accuracy = P::ACCURACY.saturated_into();
-				let value = p.deconstruct().saturated_into();
+				let accuracy = P::ACCURACY;
```

Comment:
Does this change anything? What was the type this was saturating into before? Probably dumb question but I don't know this code.

Reply:
Actually I don't understand how this every compiled. The compiler was just wildly guessing what we want here.

The crucial change here is the where clause that was added above. This makes the difference here ;)

Reply:
Not sure what was going on here either, the compiler needed to find a way to go from `PerThing` to `FixedPointOperand` 🤷